avocado/core/status/repo.py

```diff
@@ -37,6 +38,14 @@ def __init__(self, job_id):
 
     def _handle_task_finished(self, message):
         task_id = message['id']
+
+        result = message.get('result')
+        if result is not None and result.upper() not in STATUSES:
+            overriden = 'error'
+            LOG.error('Task "%s" finished message with unsupported status '
+                      '"%s", changing to "%s"', task_id, result, overriden)
+            message['result'] = overriden
```

Comment:
I'm OK with that.  Just for context, I was trying to avoid changing a test's data too much in the status repo.  IMO, the status repo should be, as much as possible, a true representation of what the test reported and not plagued with too much logic and changing data.